### ISSUE TITLE
`calcite-block`: `title` attribute obscures toggle button's accessible name in Chrome

### ISSUE THREAD

**Issue by fdeters** [Esri/calcite-components#13672]
**monday.com sync:** #10963581119

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Block

### Summary

In Google Chrome (but not Firefox), the `title` attribute on the `calcite-block`'s internal expand/collapse toggle button sets the button's accessible name to an unhelpful value. As a result, NVDA does not announce the block's descriptive name.

### Actual Behavior

In a recent accessibility audit, a tester using NVDA found that the name of each `calcite-block` was not clearly announced by the screen reader. After more investigation, here's what I've learned:

**Key points:**
- The `title` attribute messes with the accessible name of the button
- Behavior differs between browsers
- What the screen reader announces for the button differs based on whether the user is in focus mode (Tab, Shift+Tab) or browse mode (arrow keys)

**Fail state description:**
Google Chrome: When a user in **browse mode** enters the calcite block, the first node they encounter is the toggle button. NVDA announces it as "Button expanded Collapse".

When in **focus mode**, the button is announced correctly: "Collapse Button expanded [heading] [description]"

**Chrome screenshot**
See the accessible name of the toggle button is "Collapse"

![Image](https://github.com/user-attachments/assets/dcb98068-95f7-4dab-a287-4a8e59778b5b)

**Firefox screenshot**
See the accessible name of the toggle button is "[heading] [description]"

![Image](https://github.com/user-attachments/assets/76f2d5ba-551e-443c-af52-de1da595c203)

### Expected Behavior

I expect the button's announced name to be useful under all circumstances. Specifically, "Collapse Button expanded [heading] [description]" would be preferable. 

That behavior is present when using focus mode in NVDA in Chrome, and it's always present in Firefox.

### Reproduction Sample

http://codepen.io/fdeters/pen/wBWWbKb?editors=1010

### Reproduction Steps

1. Open the CodePen reproduction sample in Google Chrome
2. Open NVDA
3. Navigate to the block's toggle button using focus mode (Tab). You'll hear the proper name announced.
4. Navigate to the block's toggle button using browse mode (Arrow keys). You'll hear the unhelpful name announced.
5. (Optional) Click the `<button>` in the reproduction to remove the `title` attribute. You should hear the block's toggle button announcement change to a useful value in browse mode!

Repeat steps 3-4 in Firefox to see that the issue doesn't exist in Firefox.

I'm not able to check, but I bet JAWS would have a similar issue based on what's in the accessibility tree in the browsers.

### Reproduction Version

3.3.3

### Relevant Info

**OS:** Windows 11
**Browser:** Google Chrome Version 143.0.7499.170
**Screen reader:** NVDA 2025.3.1

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by fdeters** [comment]
Note: This is a separate issue from https://github.com/Esri/calcite-design-system/issues/11707, but I think this may have contributed to the problem described there.

**Comment by geospatialem** [comment]
Spike needed to confirm next actionable steps.
